### PR TITLE
Replace multiple calls to `add_system` with `add_systems`

### PR DESCRIPTION
**Pull request by JoJoJet**
# Objective

When multiple systems are added to an app or schedule, prefer using `add_systems`. This makes the code less repetitive, and usually results in nicer formatting.

## Example

```rust
// Before:
schedule.add_system(ab).add_system(cd).add_system(ce);

// After:
schedule.add_systems((ab, cd, ce));
```

### CODE REVIEW

**Review by hymm — APPROVED** [review]
Generally in favor of these changes. Most things are easier to read and understand. Had some comments around a few places where `chain` could have been used.

When we start applying different restrictions to multiple systems like with

```rust
.add_system((
    system1.in_set(Prepare),
    system2.in_set(Prepare),
    system3.in_set(Queue),
));
```

I think there are stylistic choices where it isn't clear what the best way to write things is. The above could also be written as:

```rust
.add_systems((system1, system2).in_set(Prepare))
.add_system(system3.in_set(Queue))
```

Probably not worth debating at this point, since we have some open issues around limitations of add_systems that could make this point moot.

File: crates/bevy_ecs/src/system/mod.rs [hunk 1]
```diff
@@ -57,10 +57,11 @@
 //! # use bevy_ecs::prelude::*;
 //! # let mut app = Schedule::new();
 //! // Prints "Hello, World!" each frame.
-//! app
-//!     .add_system(print_first.before(print_mid))
-//!     .add_system(print_mid)
-//!     .add_system(print_last.after(print_mid));
+//! app.add_systems((
```
[the remark this answers is not in the document]
Comment (JoJoJet) [reply crates/bevy_ecs/src/system/mod.rs:60]: The intention for this doctest is to be a super basic example for `.after` and `.before`, so I decided not to use `.chain()` here.

[the remark this answers is not in the document]
Comment (JoJoJet) [reply crates/bevy_ecs/src/system/mod.rs:60]: On second thought, it's probably better to show both options for ordering here.

File: crates/bevy_app/src/lib.rs [hunk 2]
```diff
@@ -197,13 +199,17 @@ impl StartupSet {
...
-        schedule.configure_set(Startup.after(PreStartupFlush).before(StartupFlush));
-        schedule.configure_set(PostStartup.after(StartupFlush).before(PostStartupFlush));
+        schedule.add_systems((
+            apply_system_buffers.in_base_set(PreStartupFlush),
+            apply_system_buffers.in_base_set(StartupFlush),
+            apply_system_buffers.in_base_set(PostStartupFlush),
+        ));
+
+        schedule.configure_sets((
+            PreStartup.before(PreStartupFlush),
```
[the remark this answers is not in the document]
Comment (hymm) [reply crates/bevy_app/src/lib.rs:209]:
```rust
schedule.configure_sets((
    PreStartup,
    PreStartupFlush,
    Startup,
    StartupFlush,
    PostStartup,
    PostStartupFlush
).chain());
```
Isn't this the same?

[the remark this answers is not in the document]
Comment (JoJoJet) [reply crates/bevy_app/src/lib.rs:209]: Ooh I like that

**Review by james7132 — APPROVED** [review]
This in theory requires a lot more codegen than the original `add_system` chain. Wonder if this has any measurable impact on `cargo build --timings`.

**Pull request merged**